### PR TITLE
Improved configurations reader to extract store configs

### PR DESCRIPTION
**Pull request by Niveathika92**

### CODE REVIEW

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java [hunk 1]
```diff
@@ -59,9 +61,39 @@ public ConfigReader generateConfigReader(String namespace, String name) {
...
+                            storeConfigs.put(SiddhiConstants.ANNOTATION_ELEMENT_TYPE, childConfiguration.getType());
+                            if (childConfiguration.getProperties() != null) {
+                                storeConfigs.putAll(childConfiguration.getProperties());
+                            }
+                            return storeConfigs;
+                        }
+                    }
+                }
+            } catch (CarbonConfigurationException e) {
+                LOGGER.error(e.getMessage(), e);
```
Comment (tishan89) [inline components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java]: Shall we add more context to the error message

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java [hunk 2]
```diff
@@ -59,9 +61,39 @@ public ConfigReader generateConfigReader(String namespace, String name) {
...
+            }
+        }
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Couldn't find a matching configuration for store, name: " +
+                    name + "!");
+        }
+        return new HashMap<>();
```
Comment (tishan89) [inline components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java]: Does the upstream code handle empty HashMap scenario?

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/RootConfiguration.java [hunk 3]
```diff
@@ -26,13 +26,17 @@
...
  */
-@Configuration(namespace = "siddhi", description = "Siddhi extension parent level configuration")
+@Configuration(namespace = "siddhi", description = "Siddhi extension and store parent level configuration")
 public class RootConfiguration {
 
     @Element(description = "Extension list", required = true)
     public List<Extension> extensions;
 
+    @Element(description = "Store list", required = false)
+    public List<Store> stores;
```
Comment (tishan89) [inline components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/RootConfiguration.java]: Any reason to use public access modifier?

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/ConfigReaderTestCase.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        String method = "POST";
+        String body = "@App:name('SiddhiApp1')\n" +
+                "@Store(ref='store5')" +
+                "define table FooTable (symbol string, price float, volume long);";
+
+        logger.info("Store reference support -  Configuring RDBMS extension");
+        HTTPResponseMessage httpResponseMessage = TestUtil.sendHRequest(body, baseURI, path, contentType, method,
+                true, "admin", "admin");
+        Assert.assertEquals(httpResponseMessage.getResponseCode(), 201);
+        Thread.sleep(10000);
```
Comment (tishan89) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/ConfigReaderTestCase.java]: Why do we need 10s sleep?

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java (hunk 2 above)
Comment (Niveathika92) [reply components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java]: Yes, it does. (https://github.com/wso2/siddhi/blob/master/modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/helper/DefinitionParserHelper.java#L833)

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/ConfigReaderTestCase.java (hunk 4 above)
Comment (Niveathika92) [reply components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/ConfigReaderTestCase.java]: This test case is no longer in the master branch

Comment (Niveathika92) [reply components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/ConfigReaderTestCase.java]: This was due to how test case was configured. Fixed in 9068071

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/RootConfiguration.java (hunk 3 above)
Comment (Niveathika92) [reply components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/RootConfiguration.java]: Fixed in 279c00f

File: components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java (hunk 1 above)
Comment (Niveathika92) [reply components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java]: Fixed in 017fa91

**Pull request merged**